### PR TITLE
Code of conduct (based on Contributor Covenant)

### PR DESCRIPTION
**Pull request by ionas**
**Explicit instead of Implicit rule set**
- An explicit instead of an implicit rule set because implicit rule sets allows for arbitrary application (far stretched up to ignored).
- Preliminary suggestion / PR (the list could be changed and/or extended).
- Note: Like the first CC based PR (#7725), this _also_ needs translation into well represented languages to take effect (else a CoC defeats itself).
- You can find the arguments for a specific list here: https://github.com/cakephp/cakephp/pull/7725#issuecomment-159018637, or here https://github.com/cakephp/cakephp/issues/7682#issuecomment-158915472 for instance.

### CODE REVIEW

**Comment by josegonzalez** [comment]
We can probably translate to Spanish and French, which should cover the majority of our users. I'd say Japanese as well if we can get someone from that side of the community to contribute.

**Comment by ionas** [comment]
Spanish and French is a good idea. Japanese, Simplified Chinese, Russian and Arabic would be good too, but as long as there are not at least two trustworthy people it is hard to do.

I could do the German translation and @dereuromark or @ravage84 could proof read that one (spirit, semantics, syntax).

**Comment by thaJeztah** [comment]
I see there's no longer an e-mail address mentioned in the document, is this something that's left to be added?

**Comment by ionas** [comment]
@theJeztah if something is missing it wasn't on purpose.

**Comment by thaJeztah** [comment]
@ionas no problem, I just noticed it wasn't there anymore, so thought I'd mention it :+1:, writing a good CoC is difficult, so thanks for starting this!

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+body size, race, ethnicity, age, religion, or nationality.
+
+In essence: **Don't be Evil**, where...
+
+* 'Evil' is: Discriminating others because of their language, gender, religion or reputation. Don't do that.
+* 'Evil' is: Disregarding works of others because of their language, gender, religion or reputation. Don't do that.
+* 'Evil' is: Harassing individuals or groups on Github, google groups or public IRC chat. Don't do that.
+* 'Evil' is: Insulting others (limited amounts of swearing are tolerated, as long as the target are dead things, not beings). Don't do that.
+* 'Evil' is: Intimidation and threats of violence. Don't do that.
+* 'Evil' is: Being impatient and demanding. Don't do that.
```
Comment (markstory) [inline CODE_OF_CONDUCT.md]: You could factor out `'Evil' is:` into line 13. Something like

> In essence: **Don't be Evil** where Evil is:

**Comment by markstory** [comment]
Ping @beporter

**Comment by beporter** [comment]
Wow, thanks @markstory for looping me in. This popping up right before the (American) Thanksgiving holiday, I would have missed this for another 4-5 days otherwise. 

A couple points: 

* The Contributor Covenant is currently being translated into multiple languages: CoralineAda/contributor_covenant#139. This has two implications:

    * Less work for us since we can leverage the forthcoming base translations from CC.

    * I feel strongly that any additional translations that we do here specifically for Cake should be "backported" to CC too. That's the honorable thing to do and in the spirit of FOSS. We have the ability and opportunity to bring the highly-diverse Cake community's talent to bear in assisting with CC's own improvement, and so we should.

* It's interesting that the CC actually had some "specific" things not to do in the previous list, that are now removed and that I don't feel are covered by the replacement lines. For example:

    ```
    The use of sexualized language or imagery.

    Publishing other's private information, such as physical or 
    electronic addresses, without explicit permission.
    ```

    Are these behaviors that we will accept as a community because they are not explicitly mentioned in this version of the CoC? I do think they are rather important ones.

* Why are the expectations of proper behavior limited only to GitHub, IRC and Google Groups?

* Suffice to say, I don't necessarily agree with having an explicit list since personally I think it's too easy to act to the letter of the CoC and not the spirit, but(!) as I've said before I'd rather have an imperfect CoC than none and I'm willing to compromise to get this in. I do not object to continuing, for whatever that's worth.

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be  
+reported by contacting a project maintainer at conduct@cakefoundation.org. All  
+complaints will be reviewed and investigated and will result in a response that  
+is deemed necessary and appropriate to the circumstances. Maintainers are  
+obligated to maintain confidentiality with regard to the reporter of an  
+incident.  
+
+
+This Code of Conduct is losely based on the [Contributor Covenant][homepage],
```
Comment (beporter) [inline CODE_OF_CONDUCT.md]: `losely` :arrow_right: `loosely`

**Comment by HavokInspiration** [comment]
From what I could read, the French translation seems well written. Once everything is settled, I can give it a more thorough look.

**Comment by ionas** [comment]
@beporter so about the semantics and missing rules.
1. The use of sexualized language or imagery.
2. Publishing other's private information, such as physical or 
electronic addresses, without explicit permission.

I did not oversee them. The reason I dropped those is
For 1.: because there is no reference to persons. You can always ignore people or just read across. A f**k or s**t or d**n word here or there is really a problem? If you think so and you think it is such a problem that it should be in a CoC, then it should be included. I'd suggest adding 'Excessive or repeated use of sexualized language or imagery"
For 2.: Because it is often required to change contact details and include third parties via CC (or even sometimes BCC) for instance. If you take a CoC SERIOUSLY then this will be prohibited unless you ask for EXPLICIT permission each time. If you have ever used Facebook or LinkedIn and allowed them to read your Address Book (of iOS or Android) you already crossed the line a hundred of times and that's a far worse miss-behavior than this CoC entry. People will still do it, and even after the CoC, and even core members will still do it as stopping this requires a big publicity agenda.

So for 1. if you think its that relevant, include it back in. For 2. I'd not like that to be included because it is both impractical and irrelevant (because people share stuff anyway without thinking twice, sadly).

**Comment by markstory** [comment]
@ionas The intent of the 'publish address' clause is to communicate that doxxing is not acceptable. Your examples imply that a person has already published some or all of the information. We are also not trying to cover scenarios related to linkedin or facebook.

My interpretation of 'sexualized language' is less about curse words and more about perjoratives often used when referring to lgbtq folks.

Like @beporter I think these are important don't s to have.

**Comment by ionas** [comment]
@markstory 
About Doxxing: The clause should then say exactly that. You are not allowed to publish personal details not otherwise publicly accessible to prohibit Doxxing and alike.

About sexualized Language: If it is about LGBTQ (and I am very aware that especially in tech communities contrary to social sciences there is a big problem with discrimination of non normative genders and personalities) this should either be explicitly mentioned. But then what about the other clauses of discrimination, don't those include this already?

Are we on par for not including the last item on the CC list?

**Comment by beporter** [comment]
> About Doxxing: The clause should then say exactly that. You are not allowed to publish personal details not otherwise publicly accessible to prohibit doxxing, swatting, spamming and alike.

Unfortunately your proposed change doesn't actually address the reality of doxing at all. Under your approach, any "publicly accessible" information is fair game for **re**-pubishing. For example, if I "publish" my home address in a DNS WHOIS record, then that's my choice, yes, but the CoC is designed to discourage _others_ from taking that "already published" information and posting it in Cake IRC or on Reddit without my permission.

The clause is still vital because the point is not that someone's information may or may not already be "out there," it's that _you_ are not allowed to decide how it is distributed, only _the owner_ is.

> About sexualized Language: If it is about LGBTQ (and I am very aware that especially in tech communities contrary to social sciences there is a big problem with discrimination of non normative genders and personalities) this should either be explicitly mentioned. But then what about the other clauses of discrimination, don't those include this already?

Excluding the last clause would mean that it would be acceptable for another straight cis white man to say the offensive example [I posted earlier](https://github.com/cakephp/cakephp/pull/7725#issuecomment-159914908) to me, also a straight cis white man. In that scenario, the phrase is not originating from, directed at or even _involves_ LGBTQ at all, but it's still patently offensive. This clause is also still vital.

---

Of course, I continue to feel that the original unmodified Contributor Covenant did a reasonably good job of covering all of these cases we're trying to "patch" around with our Don't list. Credit to @ionas for forcing us to articulate why the original CC actually *is* a pretty well crafted document though. :+1: 

I also don't think we need to rehash these same points anymore so I'm stepping back unless there's a new issue to be addressed.

**Comment by beporter** [comment]
Just to clarify that last point, what I mean is: @ionas Is right about the CC having vague language, and I wouldn't object to having a separate explanation of what the "examples" in the Contributor Covenant actually mean in practice (based on all of these conversations), but I also mean that the Contributor Covenant's language also appropriately _covers_ all of these cases we're trying to address with our modifications.

**Comment by ionas** [comment]
1) About DNS records. If you spread your information on the web, it IS there. If you don't like that to be the case, you need to have a placeholder (like someone else or a legal entity) to register a domain in your name. If you cannot do that, you need to change laws of the NIC relevant to your jurisdiction. Just because you can setup some edge cases this does not convince me that there should be the requirement to get explicit permission before posting PREVIOUSLY KNOWN PUBLICLY ACCESSIBLE personal information. Anyway, I think this point is moot. I can live with it but I still think it *hurts more than it helps* as the abusive shares will happen anyway (those who do evil won't be stopped by a CoC item) and this CoCs item will merely push it out of sight if it does anything. So :-1:, I can stand it but it makes me feel sad and I am out of this discussion point.

2) The question to me is not if the CC does cover all possible cases, a sentence like "don't be evil" would be sufficient to do that. It DOES cover ALL cases the moment you define evil.

I won't comply with the last clause of the CC as it is totally vague and a horrible new privilege given to a few then legitimated people (the people in charge of interpretation). It is a general clause that allows interpretation of almost anything someone does not like into something against the CoC. It is - as I said - part of a culture of fear and control instead of freedom and friendliness. Nothing I want to or will support.

3) Just that you understand the trend you are setting (I did invite you to further discuss the matter via mail, and I'd still like to know why you see arbitrary control by a non-legitimate higher body as a solution): If society moves towards having an always observing state with death penalty and police brutality and a lot of behavioral laws in which any non-males can walk around nipples-nude, all genders share one toilet and non-normative gender people can kiss each other in public, yet sick and poor people die on the streets, it is still a horrible non-free, non-friendly society - as Foucault put it, Humanism is for the economically privileged.

This is not where I want to see things going; I want to see privileges and powers being reduced, not create new ones in addition. A CoC that tries control and observe everything and sets up as broad behavioral rules as possible to catch any possible discrimination and THEN sets up a non-democratic privileged execution body is entirely within that train of thought - it is not the beginning of emancipation, but the end.

Final Statement: **I'd still strongly suggest to drop the last very generalist item from the list and remove the "example" word from the sentence preceding the list, as that takes away arbitrary interpretation but keeps it easy to translate and/or give back to CC.**

**Comment by chinpei215** [comment]
I disagree your suggestion. Following your thoughts, the project maintainers can ban my account permanently for no reason because there are not any explicit rules about banning of accounts. You might deny it. Even so they can suddenly push explicit rules about banning of accounts, ban my account, and revert the rules. They have arbitrary rights already because they are the project maintainers. But after getting CoC, then I can mail to conduct [at] cakefoundation.org. We contributors drop nothing but gain something.

**Comment by ionas** [comment]
There are no explicit rules afterwards either, due to the last clause about unethical and unprofessional conduct. Almost every behavior can be seen as either of these.

**Comment by chinpei215** [comment]
Disagree. The project maintainers can add any arbitrary rules any time. This is not such a wording issue. Our spirit. That's all.

**Comment by ionas** [comment]
@chinpei215 true, but there is no document pretending anything else or legitimating even that. Now there is.

**Pull request closed without merging**